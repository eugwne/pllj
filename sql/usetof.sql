CREATE OR REPLACE FUNCTION pg_temp.srf()
RETURNS SETOF integer AS $$
  coroutine.yield(1)
  coroutine.yield(nil)
  coroutine.yield(2)
  coroutine.yield()
  coroutine.yield(3)
  coroutine.yield(NULL)
  coroutine.yield(4)
$$ LANGUAGE pllju;

select quote_nullable(pg_temp.srf());

create function pg_temp.tf2()
  returns setof integer
  language pllju
  as $f$
    for i = 1,4 do coroutine.yield(i) end
$f$;

select * from generate_series(1,3) i, lateral (select pg_temp.tf2() limit i) s;

create function pg_temp.t2()
  returns setof integer
  language pllju
  as $$
    for i = 1,4 do coroutine.yield(i) end
$$;

select pg_temp.t2(), pg_temp.t2();

create type type4 as (a text, b int, c boolean);

create function pg_temp.t4() returns setof type4 as $$
    coroutine.yield("('string value',40,false)") 
    coroutine.yield("('value 2',50,)") 
    coroutine.yield("('text',,true)") 
$$  language pllju;

select x.* from pg_temp.t4() as x;

create function pg_temp.t3() returns setof type3 as $$
    coroutine.yield({{{'testvalue', 20, true}, false}, 10}) 
    coroutine.yield({a = { a = { a = 'testvalue', b = 20, c = true}, b = false}, b = 10}) 

$$  language pllju;

select pg_temp.t3();
