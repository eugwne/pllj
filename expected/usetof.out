CREATE OR REPLACE FUNCTION pg_temp.srf()
RETURNS SETOF integer AS $$
  coroutine.yield(1)
  coroutine.yield(nil)
  coroutine.yield(2)
  coroutine.yield()
  coroutine.yield(3)
  coroutine.yield(NULL)
  coroutine.yield(4)
$$ LANGUAGE pllju;
select quote_nullable(pg_temp.srf());
 quote_nullable 
----------------
 '1'
 NULL
 '2'
 NULL
 '3'
 NULL
 '4'
(7 rows)

create function pg_temp.tf2()
  returns setof integer
  language pllju
  as $f$
    for i = 1,4 do coroutine.yield(i) end
$f$;
select * from generate_series(1,3) i, lateral (select pg_temp.tf2() limit i) s;
 i | tf2 
---+-----
 1 |   1
 2 |   1
 2 |   2
 3 |   1
 3 |   2
 3 |   3
(6 rows)

create function pg_temp.t2()
  returns setof integer
  language pllju
  as $$
    for i = 1,4 do coroutine.yield(i) end
$$;
select pg_temp.t2(), pg_temp.t2();
 t2 | t2 
----+----
  1 |  1
  2 |  2
  3 |  3
  4 |  4
(4 rows)

create type type4 as (a text, b int, c boolean);
create function pg_temp.t4() returns setof type4 as $$
    coroutine.yield("('string value',40,false)") 
    coroutine.yield("('value 2',50,)") 
    coroutine.yield("('text',,true)") 
$$  language pllju;
select x.* from pg_temp.t4() as x;
       a        | b  | c 
----------------+----+---
 'string value' | 40 | f
 'value 2'      | 50 | 
 'text'         |    | t
(3 rows)

create function pg_temp.t3() returns setof type3 as $$
    coroutine.yield({{{'testvalue', 20, true}, false}, 10}) 
    coroutine.yield({a = { a = { a = 'testvalue', b = 20, c = true}, b = false}, b = 10}) 

$$  language pllju;
select pg_temp.t3();
               t3                
---------------------------------
 (,)
 ("(""(testvalue,20,t)"",f)",10)
(2 rows)

